### ISSUE TITLE
[data grid] Tab key functionality is broken when using DatePicker within the MUI DataGrid

### ISSUE THREAD

**Issue by vlastocom** [mui/mui-x#15965]
### Related page

https://mui.com/x/react-data-grid/custom-columns/#date-pickers

### Kind of issue

Broken demo

### Issue description

Well, kind of broken. Integrating MUI DatePickers this way has one downside: If you switch to editMode="row" for your grid, the tab key can no longer be used to move between the fields in the edited row. And if you select a date (dateTime) field, you cannot use the tab key to move out. I wonder if there is an easy way to fix this?

Steps to reproduce:
1. Click on the "open in CodeSandbox" button under the code for the demo in this section
2. Change line 167 to: `<DataGrid rows={rows} columns={columns} editMode="row" />`
3. Go to the demo pane
4. Double-click the first column in any row (the row will become editable)
5. Use the tab key to move to the next cell (works)
6. Use the tab key to move to the date cell after (broken)
7. Use shift-tab key to move to the first cel again (works)

### Context

I simply want to integrate DatePicker (which is much more eye-pleasing and robust to my DataGrid implementation, but have to edit my grid in the "row" mode, not the "cell" mode.

**Search keywords**: TAB DataGrid DatePicker broken

**Comment by vlastocom** [comment]
Oh... one more thing. In the above steps, if in the point 4 double-click the row in the third column, the row becomes editable, but the DateTimePicker in the fourth column becomes focused instead (unexpected) and the tab key can't be used at all to move to other editable fields.

**Comment by k-rajat19** [comment]
Hi, 
you have to handle focus manually when the cell receives focus

> Oh... one more thing. In the above steps, if in the point 4 double-click the row in the third column, the row becomes editable, but the DateTimePicker in the fourth column becomes focused instead (unexpected) and the tab key can't be used at all to move to other editable fields.

remove `autoFocus` prop

**Comment by michelengelen** [comment]
> Hi, you have to handle focus manually when the cell receives focus
> 
> > Oh... one more thing. In the above steps, if in the point 4 double-click the row in the third column, the row becomes editable, but the DateTimePicker in the fourth column becomes focused instead (unexpected) and the tab key can't be used at all to move to other editable fields.
> 
> remove `autoFocus` prop

@vlastocom could you try the provided workaround?

**Comment by LukasTy** [comment]
Having an `autoFocus` set with `editMore="row"` does not sound optimal.
As for when this prop is removed, I think there are still some problems with navigation, could @mui/xgrid check this? 🤔 

https://github.com/user-attachments/assets/463d886d-d675-444f-91ff-73cbebacc0ac

**Comment by vlastocom** [comment]
> > Hi, you have to handle focus manually when the cell receives focus
> > > Oh... one more thing. In the above steps, if in the point 4 double-click the row in the third column, the row becomes editable, but the DateTimePicker in the fourth column becomes focused instead (unexpected) and the tab key can't be used at all to move to other editable fields.
> > 
> > 
> > remove `autoFocus` prop
> 
> @vlastocom could you try the provided workaround?

When you remove autofocus property rom both componets, the behaviour does not change.

**Comment by vlastocom** [comment]
After trying the above, I have tried an additional step after removing autofocus prop, which is the change to the WrappedFridEditDateInput function to handle the focus "manually":

```
function WrappedGridEditDateInput(props: TextFieldProps) {
  const { InputProps, focused, ...other } = props;
  return (
    <GridEditDateInput 
      fullWidth 
      {...InputProps}
      {...(other as InputBaseProps)}
      inputRef={input=>input&&input.focus()}
      />
  );
}
```

(inputRef prop is new in the above code, the rest is just a reformatted original example).

The effect is this:
(+) If I click on the column 1 and 2, I can successfully tab through all four columns
(-) Shift-Tab does not work on date / datetime columns (i.e. I cannot go back)
(-) The last column (the datetime one) will get focus when the row is selected

There is still something fundamentally missing there.

**Comment by k-rajat19** [comment]
Hi @vlastocom,
I've created a working demo here:  https://stackblitz.com/edit/react-qghbduzu?file=Demo.tsx 
Does this work for you?

**Comment by vlastocom** [comment]
My word! Yes, this works. Thousand thanks!!! Incidentally, somewhere in the process I tried something like this, but I used the useEffect instead of the unstable_useEnhancedEffect (need to read upon that) and I might have chosen the wrong ref to call...

Thank you very much for providing a neat solution!

**Comment by michelengelen** [comment]
Thanks @k-rajat19 for proividing a great solution for this problem.
The linked PR looks good, but needs a second review.
